editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/util/maven/MavenUtils.java

```diff
@@ -181,4 +183,25 @@ public static void alignFuseRuntimeVersion(Model mavenModel, String camelVersion
 		}
 	}
 
+	/**
+	 * If staging repositories enabled, ensure to have them in the Maven model.
+	 * 
+	 * @param mavenModel
+	 * 
+	 * @return if a modification has been done on the model
+	 */
+	public static boolean manageStagingRepositories(Model mavenModel) {
+		if(new StagingRepositoriesPreferenceInitializer().isStagingRepositoriesEnabled()){
+			boolean hasBeenUpdated = false;
+			for(List<String> nameURlPair : new StagingRepositoriesPreferenceInitializer().getStagingRepositories()){
+				String repoURI = nameURlPair.get(1);
+				hasBeenUpdated |= MavenUtils.ensureRepositoryExists(mavenModel.getRepositories(), repoURI, nameURlPair.get(0));
+				hasBeenUpdated |= MavenUtils.ensureRepositoryExists(mavenModel.getPluginRepositories(), repoURI, nameURlPair.get(0));
+			}
+			return hasBeenUpdated;
+		} else {
+			return false;
+		}
+	}
+
```

Comment:
I created the PR on the initial project (and will be fixed in this PR too of course)  https://github.com/fabric8-quickstarts/spring-boot-camel-xml/pull/125